### PR TITLE
[LMS-448] Support for Harmonizing Users in PostgreSQL

### PR DESCRIPTION
**Pull request by stephenfuqua**
Working with Canvas. Will be easy to get the next two LMS's.

The program will error out on sections. However, I was able insert fake `lms.LMSUser` and `edfi.Student` and see that the `LMSUser`'s Ed-Fi ID was set successfully.

```sql
   insert into edfi.student
           (personaltitleprefix
           ,firstname
           ,middlename
           ,lastsurname
           ,generationcodesuffix
           ,maidenname
           ,birthdate
           ,birthcity
           ,birthstateabbreviationdescriptorid
           ,birthinternationalprovince
           ,birthcountrydescriptorid
           ,dateenteredus
           ,multiplebirthstatus
           ,birthsexdescriptorid
           ,citizenshipstatusdescriptorid
           ,studentuniqueid
           ,createdate
           ,lastmodifieddate
           ,id)
     values
           (NULL
           ,'FirstName'
           ,NULL
           ,'LastName'
           ,NULL
           ,NULL
           ,'2010-01-01 00:00:00'
           ,NULL
           ,NULL
           ,NULL
           ,NULL
           ,NULL
           ,NULL
           ,NULL
           ,NULL
           ,'{student_unique_id}'
           ,'2021-01-01 00:00:00'
           ,'2021-01-01 00:00:00'
           ,'00000000-0000-0000-0000-000000000001'
           );


      insert into lms.lmsuser
           (sourcesystemidentifier
           ,sourcesystem
           ,userrole
           ,sisuseridentifier
           ,localuseridentifier
           ,name
           ,emailaddress
           ,sourcecreatedate
           ,sourcelastmodifieddate
           ,createdate
           ,lastmodifieddate
           ,deletedat)
     values
           ('sssssssssssssss'
           ,'Canvas'
           ,'{USER_ROLE}'
           ,'{student_unique_id}'
           ,'{sis_identifier}2'
           ,'{sis_identifier}3'
           ,'{email_address}'
           ,NULL
           ,NULL
           ,'2021-01-01 00:00:00'
           ,'2021-01-01 00:00:00'
           ,NULL
           )
```

### CODE REVIEW

File: src/lms-ds-loader/.env.example [hunk 1]
```diff
@@ -8,16 +8,15 @@ DB_SERVER=localhost
 
 #DB_ENGINE=mssql
 #ENCRYPT_SQL_CONNECTION=True
+# Use with caution! Primarily useful when testing against a self-signed certificate
```
Comment (stephenfuqua) [inline src/lms-ds-loader/.env.example:11]: Improvements in the DS Loader's example file.

File: src/lms-harmonizer/edfi_lms_harmonizer/helpers/argparser.py [hunk 2]
```diff
@@ -202,20 +224,12 @@ def parse_main_arguments(args_in: List[str]) -> MainArguments:
...
-        if args_parsed.useintegratedsecurity:
-            arguments.build_mssql_adapter_with_integrated_security()
-        else:
-            arguments.build_mssql_adapter(
-                args_parsed.username,
-                args_parsed.password,
-            )
-    else:
-        arguments.build_postgresql_adapter(args_parsed.username, args_parsed.password)
-
```
Comment (stephenfuqua) [inline src/lms-harmonizer/edfi_lms_harmonizer/helpers/argparser.py:220]: This older code was creating the adapter as soon as the arguments are parsed. I wanted to throw an exception if username and password are missing. However, the logging is not setup (in `__main__`) until _after_ argument parsing. Thus I moved the adapter initialization into the `get_adapter` function for lazy loading.

File: src/lms-harmonizer/edfi_lms_harmonizer/helpers/argparser.py [hunk 3]
```diff
@@ -62,12 +73,23 @@ def build_mssql_adapter_with_integrated_security(self) -> None:
...
+    def build_postgresql_adapter(self) -> None:
+        self._assert_credentials(self.username, self.password)
+
         self.adapter = sql_adapter.create_postgresql_adapter(
-            username, password, self.server, self.db_name, self._get_pgsql_server_port()
+            self.username, self.password, self.server, self.db_name, self._get_pgsql_server_port()
         )
 
     def get_adapter(self) -> sql_adapter.Adapter:
+        if not hasattr(self, 'adapter'):
```
Comment (stephenfuqua) [inline src/lms-harmonizer/edfi_lms_harmonizer/helpers/argparser.py:84]: This is an interesting side effect of having a dataclass.  It throws an error _when reading an unset attribute_ that is declared in `__post_init__`. Workaround: `hasattr`. Note that we're still able to _set_ the attribute, thanks to `__post_init__`.

**Review by bradbanister — APPROVED** [review]
Looking good

**Pull request merged**